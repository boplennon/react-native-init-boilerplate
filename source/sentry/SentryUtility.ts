import { Sentry } from 'react-native-sentry';
import _ from 'lodash';
import CrossConfigReader from '../config/CrossConfigReader';

/**
 * Utility for Sentry bug tracking (@see https://sentry.io/)
 *
 * Will configure tracking based on "config.expo.extra.sentryPublicDsn" expo configuration in app.json
 *
 * @class SentryUtility
 */
class SentryUtilityInternal {
  install() {
    if (_.isNil(Sentry)) {
      console.log(
        '** Warning: Sentry install called but module not found (SentryUtility) **'
      );
      return;
    }
    try {
      // Set up sentry bug tracking
      Sentry.config(
        CrossConfigReader.GetEnv().SENTRY_URI
      ).install();
      console.log('** Sentry installed and initiated (SentryUtility) **');
    } catch (e) {
      console.log(e);
    }
  }
}

// Global instance
const SentryUtility = new SentryUtilityInternal();
export default SentryUtility;
