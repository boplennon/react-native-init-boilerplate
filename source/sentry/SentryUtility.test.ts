import SentryUtility from './SentryUtility';

describe('components', () => {
  describe('SentryUtility', () => {
    it('should match snapshot', () => {
      expect(SentryUtility).toMatchSnapshot();
    });

    it('install should match snapshot', () => {
      expect(SentryUtility.install()).toMatchSnapshot();
    });
  });
});
