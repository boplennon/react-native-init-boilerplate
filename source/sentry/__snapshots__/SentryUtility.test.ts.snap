// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SentryUtility install should match snapshot 1`] = `undefined`;

exports[`components SentryUtility should match snapshot 1`] = `SentryUtilityInternal {}`;
